Ginger/Ginger/BusinessFlowPages_New/AddActionMenu/RecordNavPage.xaml.cs

```diff
@@ -40,197 +27,289 @@ namespace Ginger.BusinessFlowsLibNew.AddActionMenu
+        /// Constructor
+        /// </summary>
+        /// <param name="context"></param>
         public RecordNavPage(Context context)
         {
             InitializeComponent();
             mContext = context;
             xWinGridUC.mContext = mContext;
+            xWinGridUC.WindowsComboBox.SelectionChanged += WindowsComboBox_SelectionChanged;
+
+            InitMethods();
+            context.PropertyChanged += Context_PropertyChanged;
         }
 
-        private void RecordingButton_Click(object sender, RoutedEventArgs e)
+        private void InitMethods()
+        {
+            SetPOMControlsVisibility(false);
+            SetControlsDefault();
+            SetMultiplePropertiesGridView();
+            SetDefault(mContext);
+        }
+
+        private void Context_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            if(e != null && e.PropertyName == nameof(BusinessFlow))
+            {
+                mContext = (Context)sender;
+                xWinGridUC.mContext = mContext;
+                InitMethods();
+                if (IsRecording)
+                {
+                    IsRecording = !IsRecording;
+                    StopRecording();
+                    SetRecordingButtonText();
+                }
+            }
+        }
+
+        /// <summary>
+        /// This method will set the page to default
+        /// </summary>
+        /// <param name="context"></param>
+        public void SetDefault(Context context)
+        {
+            if(mActParentActivity != context.BusinessFlow.CurrentActivity)
+            {                
+                xWinGridUC.UpdateWindowsList();
+                xIntegratePOM.IsChecked = false;
+            }
+            mActParentActivity = context.BusinessFlow.CurrentActivity;
+
+            if (AgentHelper.Instance.CheckIfAgentIsRunning(context))
+            {
+                mWindowExplorerDriver = AgentHelper.Instance.WindowExplorerDriver;
+                xWinGridUC.mWindowExplorerDriver = mWindowExplorerDriver;
+                xStartAgentButton.IsEnabled = false;
+            }
+            else
+            {
+                xStartAgentButton.IsEnabled = true;
+            }
+        }
+
+        public void SetMultiplePropertiesGridView()
         {
-            isRecording = !isRecording;
-            UpdateUI();
-            //stop Live Spy
-            //if (LiveSpyButton.IsChecked == true)
-            //    if (dispatcherTimer != null)
-            //        dispatcherTimer.IsEnabled = false;
+            gridPOMListItems.SetTitleLightStyle = true;
+            GridViewDef view = new GridViewDef(GridViewDef.DefaultViewName);
+            view.GridColsView = new ObservableList<GridColView>();
+            view.GridColsView.Add(new GridColView() { Field = nameof(POMBindingObjectHelper.ItemName), Header = "Name", WidthWeight = 120, AllowSorting = true, BindingMode = BindingMode.OneWay, ReadOnly = true });
```

Comment:
no need name and containing folder, we need one field showing all POM path and name, same like we have in POM mapping in UIElement action

Reply:
Done